docs/developer-docs/integrations/sns/managing-sns/sns-asset-canister.md

```diff
@@ -0,0 +1,162 @@
+:::
+
+## Deploying an asset canister
+
+To deploy an asset canister, the `./deploy_assets.sh` script found in the [SNS testing repo](https://github.com/dfinity/sns-testing) can be used.
+
+Then, the SNS can be deployed using the `./deploy_sns.sh <config-path>` script. After the SNS has been deployed, the following steps can be used to prepare the SNS canister for being handed off to the SNS:
+
+:::info
+These steps should be run before the `open_sns_swap.sh` script is run according to the SNS lifecycle section.
+:::
+
+- #### Step 1: To prepare the asset canister, run the following script:
+
+```
+./prepare_assets.sh
+```
+
+This grants the SNS governance canister the necessary permissions to manage access to the asset canister. 
+
+- #### Step 2: Next, register the asset canister with the SNS by running the following script:
+
+```
+./register_dapp.sh <canister-id>
+```
+
+where `<canister-id>` is the canister ID of the asset canister.
+
+- #### Step 3: Then, register the SNS's managing permissions. 
+To test managing permissions in the asset canister via SNS proposals, you need to register the asset canister's functions to manage permissions with the SNS as generic functions. This is accomplished by running the following script:
+
+```
+./register_permission_assets.sh.
+```
+
+Now you can add or revoke a permission to the asset canister via SNS proposals. To this end, run the script:
+
+```
+./add_permission_assets.sh <dfx-identity> <permission>
+```
+
+or 
+
+```
+./revoke_permission_assets.sh <dfx-identity> <permission>
+```
+
+respectively, where `<dfx-identity>` is the name of an identity to manage permissions for and `<permission>` is one of Commit, Prepare, and ManagePermissions. You can check the actual permissions by running:
+
+```
+dfx canister --network ${NETWORK} call assets list_permitted '(record {permission = variant {<permission>}})'
+```
+
+or by running the script:
+
+```
+./get_permission_assets.sh.
+```
+
+- #### Step 4: Make sure that you can still commit assets by running the script:
+
+```
+./commit_assets.sh <path> <content>
+```
+
+where `<path>` is the HTTP path of the asset, e.g., `/myasset.txt` and `<content>` is the ASCII-encoded content of the asset.
+
+
+- #### Step 5: Testing all possible SNS launch scenarios includes testing a failed swap (e.g., if not enough funds have been raised) where the control of the asset canister is given back to your principal.
+
+You should run the following step after `finalize_sns_swap.sh` for an unsuccessful swap:
+
+   - After a failed swap, you can run the script:
+   
+   ```
+   ./take_ownership_assets.sh
+   ```
+   
+   to reset the permissions of the asset canister back to only your principal having the Commit permission (in particular, with the SNS governance having no permission anymore).
+
+
+
+
+## Submitting a proposal 
```

Comment:
can you specify if its an SNS proposal or an NNS proposal (I myself do not know)